tests/manage/pv_services/test_run_io_multiple_pods.py

```diff
@@ -1,103 +1,54 @@
 
 
-@pytest.fixture()
-def init_pvc_size(request):
-    """
-    Initialize the PVC size for PVC creation
-    """
-    class_instance = request.node.cls
-    class_instance.pvc_size_int = getattr(
-        class_instance, 'pvc_size_int', random.randint(1, 10)
-    )
-    class_instance.pvc_size = f'{class_instance.pvc_size_int}Gi'
-
-
 @tier2
-@pytest.mark.usefixtures(
-    create_project.__name__,
-    init_pvc_size.__name__,
-)
-class BaseRunIOMultiplePods(ManageTest):
+class TestIOMultiplePods(ManageTest):
     """
     Run IO on multiple pods in parallel
     """
     num_of_pvcs = 10
-    pvc_size_int = 5
-    interface = None
-
-    def run_io_multiple_pods(self):
-        """
-        Run IO on multiple pods in parallel
-        """
-
-        for pod in self.pod_objs:
-            pod.run_io('fs', f'{self.pvc_size_int - 1}G')
-
-        for pod in self.pod_objs:
-            fio_result = pod.get_fio_results()
-            logger.info(f"IOPs after FIO for pod {pod.name}:")
-            logger.info(
-                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+    pvc_size = 5
+
+    @pytest.fixture(
+        params=[
+            pytest.param(
+                constants.CEPHBLOCKPOOL, marks=pytest.mark.polarion_id("OCS-692")
+            ),
+            pytest.param(
+                constants.CEPHFILESYSTEM, marks=pytest.mark.polarion_id("OCS-693")
             )
-            logger.info(
-                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
-            )
-
-
-@tier2
-@pytest.mark.polarion_id("OCS-692")
-@pytest.mark.usefixtures(
-    create_rbd_secret.__name__,
-    create_ceph_block_pool.__name__,
-    create_rbd_storageclass.__name__,
-    create_pvcs.__name__,
-    create_pods.__name__
-)
-class TestRunIOMultiplePodsRBD(BaseRunIOMultiplePods):
-    """
-    Run IO on multiple pods in parallel - RBD
-    """
-    interface = constants.CEPHBLOCKPOOL
-    storage_type = 'block'
-
-    def test_run_io_multiple_pods_rbd(self):
-        """
-        Run IO on multiple pods in parallel - RBD
+        ]
+    )
+    def pods(self, request, pod_factory, pvc_factory):
         """
-        self.run_io_multiple_pods()
+        Prepare multiple pods for the test
 
+        Returns:
+            list: Pod instances
 
-@tier2
-@pytest.mark.polarion_id("OCS-693")
-@pytest.mark.usefixtures(
-    create_cephfs_secret.__name__,
-    create_cephfs_storageclass.__name__,
-    create_pvcs.__name__,
-    create_pods.__name__
-)
-class TestRunIOMultiplePodsFS(BaseRunIOMultiplePods):
-    """
-    Run IO on multiple pods in parallel - CephFS
-    """
-    interface = constants.CEPHFILESYSTEM
-    storage_type = 'fs'
-
-    def test_run_io_multiple_pods_fs(self):
         """
-        Run IO on multiple pods in parallel - CephFS
+        custom_pvc = pvc_factory(size=self.pvc_size)
+        pods = [
+            pod_factory(
+                interface=request.param, pvc=custom_pvc
```

Comment:
If I understood correctly, Isn't this trying to attach the same pvc to all the pods, which will cause the test to fail.
Is this script passing?

Reply:
Great catch!